src/middlewares/createRouterMiddleware.ts

```diff
@@ -62,13 +62,16 @@ export const createRouterMiddleware = <
 
     const initialUiState = instantSearchInstance._initialUiState;
 
+    let unsubscribed = false;
```

Comment:
unsubscribed needs to be set to true on `subscribe`

Reply:
we'd probably also need a test with the flow:

start
change (url changes)
dispose
change (url doesn't change)
start
change (url changes)

Reply:
> unsubscribed needs to be set to true on `subscribe`

You mean, `unsubscribed = false;` on `subscribe`?

Reply:
yes, I meant false indeed, although we probably want to make the logic opposite to avoid negations (using subscribed = false initially, and checking if !subscribed )

Reply:
Thanks, I'll make the update. Also, do we want to prevent the calls to `subscribe`/`unsubscribe` based on the `subscribed` boolean?

```
subscribe() {
  if(subscribed) return;
  subscribed = true;
   // ...
}
```

Reply:
subscribe will not be called if already subscribed, because it would mean the middleware was added twice, or InstantSearch started twice, both aren't possible

Reply:
Thanks for the clarification!